examples/attitude_example.cpp

```diff
@@ -0,0 +1,150 @@
+ * @brief Basic example script demonstrating how to use the mjbots_control_loop
+ *        to read attitude data.
+ * @date 7/9/22
+ * 
+ * @copyright 2022 The Trustees of the University of Pennsylvania. All rights reserved.
+ * 
+ */
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+#include <iostream> // cout
+#include <iomanip> // setprecision, fixed
+#include "kodlab_mjbots_sdk/mjbots_control_loop.h"
+#include "kodlab_mjbots_sdk/joint_moteus.h"
+#include "kodlab_mjbots_sdk/lcm_subscriber.h"
+#include "kodlab_mjbots_sdk/attitude.h"
+#include "AttitudeLog.hpp"
+
+class AttitudeExample : public kodlab::mjbots::MjbotsControlLoop<AttitudeLog,
+                                                                 VoidLcm>
+{
+public:
+  /**
+   * @brief Construct a new Attitude Example object
+   * 
+   * @param joints vector of robot joint shared pointers
+   * @param options options defining the behavior
+   */
+  AttitudeExample(std::vector<kodlab::mjbots::JointMoteus> joints,
+                  const kodlab::mjbots::ControlLoopOptions &options)
+      : MjbotsControlLoop::MjbotsControlLoop(joints, options),
+        att_(robot_->GetAttitudeSharedPtr()),
+        att_read_only_(robot_->GetAttitude())
+  {
+    // Modify world offset rotation to \f$\pi\f$ radians about the roll axis
+    Eigen::Quaternionf world_offset = Eigen::Quaternionf(
+      Eigen::AngleAxisf(M_PI, Eigen::Vector3f::UnitX()));
+    att_->set_world_offset(world_offset);
+    att_read_only_.set_world_offset(world_offset);
+  }
+
+private:
+  /**
+   * @brief Shared pointer to an attitude object for storing IMU data
+   * @note Uses a modified world offset rotation of \f$\pi\f$ radians about the
+   *       roll axis
+   * @note This is is a shared pointer to a continually updated `Attitude` 
+   *       object.  There is no need to update in the contorl loop itself.
+   */
+  std::shared_ptr<kodlab::Attitude<float>> att_;
+
+  /**
+   * @brief Read-only attitude object for storing IMU data
+   * @note Uses a modified world offset rotation of \f$\pi\f$ radians about the
+   *       roll axis
+   * @note Unlike `att_` above which is a shared pointer to a continually 
+   *       updated `Attitude` object, this object is only updated when 
+   *       specifically assigned.
+   */
+  kodlab::Attitude<float> att_read_only_;
+
+  /**
+   * @brief Gathers attitude data and sets command torques to zero
+   */
+  void CalcTorques() override
+  {
+    // Get Updated Read-Only Attitude
+    att_read_only_ = robot_->GetAttitude();
+    
+    // Get Updated Attitude Representations
+    Eigen::Quaternionf quat = att_->get_att_quat();
+    kodlab::rotations::EulerAngles euler = att_->get_att_euler();
+    Eigen::Matrix3f rot_mat = att_->get_att_rot_mat();
+
+    // Print Attitude Representations to Terminal
+    std::cout << "att_:" << std::endl;
+    att_->PrintAttitude();
+    std::cout << "att_read_only_:" << std::endl;
+    att_read_only_.PrintAttitude();
```

Comment:
These currently should give different answers based on the world offset bug, if not then maybe there is something more insidious

Reply:
Not a bug in world offset, `att_read_only_` is copying the `att_` shared pointer, which has the appropriate offset already set.